lua/draw.lua

```diff
@@ -19,20 +21,122 @@ function draw_ap()
 	)
 
 	print(
-		"ap: " .. curr_ap .. "/" .. max_ap,
+		"ap:" .. curr_ap .. "/" .. max_ap,
 		ap_top_left_x + 1,
 		ap_top_left_y + 1,
 		0
 	)
 end
 
+function draw_capture()
+	local mouse_x = stat(32)
+	local mouse_y = stat(33)
+
+	if capturing then
+		for agent in all(agents) do
+			local top_left_x = agent.x - 1
+			local top_left_y = agent.y - 1
+			local bot_right_x = agent.x + 1
+			local bot_right_y = agent.y + 1
+
+			if top_left_x < min_x or top_left_x == nil then
+				top_left_x = min_x
+			end
+			if top_left_y < min_y or top_left_y == nil
+				then top_left_y = min_y
+				end
+			if bot_right_x > max_x or bot_right_x == nil then
+				bot_right_x = max_x
+			end
+			if bot_right_y > max_y  or bot_right_y == nil then
+				bot_right_y = max_y
+			end
+
+			if 		mouse_x >= top_left_x
+				and mouse_x <= bot_right_x
+				and mouse_y >= top_left_y
+				and mouse_y <= bot_right_y
+			then
+				agent_to_capture = agent
+				circ(
+					agent.x,
+					agent.y,
+					4,
+					10
+				)
+				break
+			end
+		end
+	end
+end
+
+function draw_trace_lines()
+	for pair in all(trace_pairs) do
+		make_particle_line(
+			pair[1].x,
+			pair[1].y,
+			pair[2].x,
+			pair[2].y,
+			1,
+			3,
+			11,
+			trace_particle_chance
+		)
+	end
+end
+
+function draw_trace()
+	local mouse_x = stat(32)
+	local mouse_y = stat(33)
+
+	if tracing then
+		for agent in all(agents) do
+			local top_left_x = agent.x - 1
+			local top_left_y = agent.y - 1
+			local bot_right_x = agent.x + 1
+			local bot_right_y = agent.y + 1
+
+			if top_left_x < min_x or top_left_x == nil then
+				top_left_x = min_x
+			end
+			if top_left_y < min_y or top_left_y == nil
+				then top_left_y = min_y
+				end
+			if bot_right_x > max_x or bot_right_x == nil then
+				bot_right_x = max_x
+			end
+			if bot_right_y > max_y  or bot_right_y == nil then
+				bot_right_y = max_y
+			end
+
+			if 		mouse_x >= top_left_x
+				and mouse_x <= bot_right_x
+				and mouse_y >= top_left_y
+				and mouse_y <= bot_right_y
+				and agent.is_captured
```

Comment:
Same problems with unnecessary nil checks here. Since this code is identical (except for the last condition) to corresponding section in draw_capture it should probably also be extracted to its own function.